### PR TITLE
Optmizations for `ProtectApiClient`

### PR DESCRIPTION
**Pull request by AngellusMortis**
* Increases Bootstrap poll interval to 15 minutes if WS is activate. Does not need to be polled every 60 seconds
* Adds arguments to ignore a number of events from WS
* Adds test to benchmark processing Websocket packets
* Optimizes `update_from_dict` to not call `.copy` or `.construct`.

No optimization:
image

With optimizations: 
image

second pass optimizations: 
image

The `test_ws_filtered` is closer to what the HA performance gains should be. The only difference there is that NVR messages were ignored for that test to improve coverage. 

About a ~35% performance increase on the mean process time for the `update_from_dict` optimizations and then an additional 25% (52% total) increase filtering out WS messages.

### CODE REVIEW

File: pyunifiprotect/data/bootstrap.py [hunk 1]
```diff
@@ -147,6 +164,15 @@ def process_ws_packet(self, packet: WSPacket) -> Optional[WSSubscriptionMessage]
...
                 )
             if model_type in ModelType.bootstrap_models() or model_type == ModelType.EVENT.value:
+
+                if model_type == ModelType.CAMERA.value and ignore_stats:
+                    for key in list(data.keys()):
+                        if key in STATS_KEYS:
+                            del data[key]
+                    # nothing left to process
+                    if data == {}:
+                        return None
```
Comment (bdraco) [inline pyunifiprotect/data/bootstrap.py:174]: Similar code to line 148, could be a function

Comment (AngellusMortis) [reply pyunifiprotect/data/bootstrap.py:174]: Was thinking about it, but not sure since it was only 3 lines (cannot really break out the return if).

Comment (bdraco) [reply pyunifiprotect/data/bootstrap.py]: Maybe 
```

if _prune_stats_keys(data):
    return None

....

def _prune_stats_keys(data):
    if STATS_KEYS.intersection(data):
        for key in list(data):   
            if key in STATS_KEYS:
                del data[key]
    # nothing left to process
    return data == {}
```

Comment (AngellusMortis) [reply pyunifiprotect/data/bootstrap.py]: I refactored it a bit. There are only 3 keys in that set, so I do not think getting fancy with set logic will give us any real boost here. The number of keys from the WS messages is also pretty limited as well (~3 max top level, ~6 max if you include the nested stats keys that are already being stripped out as a result).

Comment (AngellusMortis) [reply pyunifiprotect/data/bootstrap.py]: Added it anyways. haha.

**Comment by bdraco** [comment]
Screen Shot 2021-12-02 at 12 42 27


Looks like this now

**Comment by AngellusMortis** [comment]
So that is a pretty good improvement. That is without filtering any of the messages as well too though since that requires a change to how we initialize the client on the HA side.

**Comment by AngellusMortis** [comment]
Added a commit on the integration branch to use the new options to API Client.

**Comment by bdraco** [comment]
Screen Shot 2021-12-02 at 15 43 00

**Comment by bdraco** [comment]
On another instance

Screen Shot 2021-12-02 at 15 49 24

**Comment by bdraco** [comment]
Testing on the original instance  https://github.com/briis/unifiprotect/pull/354#issuecomment-984219175

Original was ~47% of the event loop cpu time
Optimized version is ~25% of the event loop cpu time

**Review by bdraco — APPROVED** [review]

**Comment by bdraco** [comment]
Hopefully the filtering will get us back closer to the original implementation performance which was between 4-6%  on that instance

**Pull request merged**